apps/server/src/modules/deletion/repo/scope/deletion-request-scope.ts

```diff
@@ -9,21 +9,19 @@ export class DeletionRequestScope extends Scope<DeletionRequestEntity> {
 		return this;
 	}
 
-	byStatus(fifteenMinutesAgo: Date): this {
-		this.addQuery({
-			$or: [
-				{ status: StatusModel.FAILED },
-				{
-					$and: [{ status: [StatusModel.REGISTERED, StatusModel.PENDING] }, { updatedAt: { $lt: fifteenMinutesAgo } }],
-				},
-			],
-		});
+	byStatusAndDate(status: StatusModel[], olderThan?: Date, newerThan?: Date): this {
+		let query = { status: { $in: status } };
+		if (olderThan) {
+			const olderThanQuery = { updatedAt: { $lt: olderThan } };
+			query = { ...query, ...olderThanQuery };
+		}
 
-		return this;
-	}
+		if (newerThan) {
+			const newerThanQuery = { updatedAt: { $gte: newerThan } };
+			query = { ...query, ...newerThanQuery };
```

Comment:
This would lead to `updatedAt` being overwritten when both are specified - `olderThan` and `newerThan`. Is this really intended?

Reply:
should add as an and condition, will check it

Reply:
fixed